### PR TITLE
Publish new package `@adeira/jest-disallow-console`

### PR DESCRIPTION
**Pull request by mrtnzlml**
This is a followup after https://github.com/facebook/jest/issues/11456. Basically, our previous solution stopped working reliably and after a discussion with Jest team I realized that it's an ugly hack and we are lucky it worked for so long.

This change introduces a new `@adeira/jest-disallow-console` package which helps with testing console warnings.

The code is heavily inspired by `warnings.js` from `facebook/relay` (see: https://github.com/facebook/relay/blob/57dde664137a221a4633d1c02592b2a4e17d3feb/packages/relay-test-utils-internal/warnings.js) but it's modified to work with `global.console` instead of `warning` module.

TODO:

- [ ] close https://github.com/facebook/jest/issues/11456 after merging

### CODE REVIEW

**Review by michalsanger — COMMENTED** [review]
Do I understand correctly that unexpected `console.warn` does not fail the test run now?

File: src/jest-disallow-console/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        // eslint-disable-next-line no-console
+        console.error(`Unexpected Warning: ${message}`);
+        throw new Error(`Warning: ${message}`);
+      }
+    });
+
+  afterEach(() => {
+    if (mockExpectedWarnings.length > 0) {
+      const error = new Error(
+        `Some expected warnings where not triggered:
```
[the remark this answers is not in the document]
Comment (mrtnzlml) [reply src/jest-disallow-console/index.js]: Resolved, can you please have a look again?

**Review by michalsanger — APPROVED** [review]

**Comment by mrtnzlml** [comment]
> Do I understand correctly that unexpected `console.warn` does not fail the test run now?

I missed this comment. Yes, it means that the unexpected comments would no longer fail the test. Originally, I was thinking that it about programmers' hygiene, however, thinking about it now, these warnings can occur as we are upgrading dependencies and it would be too easy and unfortunate to miss them.

I am going to move this PR into a draft and think about it a bit better. Perhaps some mixture of these 2 approaches could work well. 🤔

**Comment by mrtnzlml** [comment]
This is not going anywhere. Our current solution is good enough and I cannot figure out how to make this work at least on the same level. Closing for now.

**Pull request closed without merging**